webapp/channels/src/components/suggestion/channel_mention_provider.tsx

```diff
@@ -14,73 +14,55 @@ import store from 'stores/redux_store.jsx';
 
 import {Constants} from 'utils/constants';
 
-import Provider from './provider';
-import Suggestion from './suggestion.jsx';
+import Provider, {ResultsCallback} from './provider';
+import {SuggestionContainer, SuggestionProps} from './suggestion';
 
 export const MIN_CHANNEL_LINK_LENGTH = 2;
 
-export type Results = {
-    matchedPretext: string;
-    terms: string[];
-    items: WrappedChannels[];
-    component: React.ElementType;
-}
-
-type ResultsCallback = (results: Results) => void;
-
-type WrappedChannels = {
+type WrappedChannel = {
     type: string;
     channel?: Channel;
     loading?: boolean;
 }
 
-export class ChannelMentionSuggestion extends Suggestion {
-    render() {
-        const isSelection = this.props.isSelection;
-        const item = this.props.item;
-        const channelIsArchived = item.channel.delete_at && item.channel.delete_at !== 0;
-
-        const channelName = item.channel.display_name;
-        let channelIcon;
-        if (channelIsArchived) {
-            channelIcon = (
-                <span className='suggestion-list__icon suggestion-list__icon--large'>
-                    <i className='icon icon-archive-outline'/>
-                </span>
-            );
-        } else {
-            channelIcon = (
-                <span className='suggestion-list__icon suggestion-list__icon--large'>
-                    <i className={`icon icon--no-spacing icon-${item.channel.type === Constants.OPEN_CHANNEL ? 'globe' : 'lock'}`}/>
-                </span>
-            );
-        }
-
-        let className = 'suggestion-list__item';
-        if (isSelection) {
-            className += ' suggestion--selected';
-        }
-
-        const description = '~' + item.channel.name;
-
-        return (
-            <div
-                className={className}
-                onClick={this.handleClick}
-                onMouseMove={this.handleMouseMove}
-                {...Suggestion.baseProps}
-            >
-                {channelIcon}
-                <div className='suggestion-list__ellipsis'>
-                    <span className='suggestion-list__main'>
-                        {channelName}
-                    </span>
-                    {description}
-                </div>
-            </div>
+export const ChannelMentionSuggestion = React.forwardRef<HTMLDivElement, SuggestionProps<WrappedChannel>>((props, ref) => {
+    const item = props.item;
```

Comment:
I'm personally less a fan of destructuring like that since I like knowing that `item` comes from props, so I had just left these as is, but I can go through and change that for consistency